File: C++/ProcessStringWithSpecialOperation.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

class ProcessStringWithSpecialOperation {
public:
    string processStr(string s) {
        string result;

        for (char ch : s) {
            if (ch >= 'a' && ch <= 'z') {
                result.push_back(ch);
            }
            else if (ch == '*') {
                if (!result.empty())
                    result.pop_back();
            }
            else if (ch == '#') {
                result += result;
            }
            else if (ch == '%') {
                reverse(result.begin(), result.end());
            }
        }

        return result;
    }
};

int main() {
    ProcessStringWithSpecialOperation sol;

    string s = "a#b%*";

    cout << "Input: " << s << endl;
    cout << "Output: " << sol.processStr(s) << endl;

    return 0;
}

File: Python/ProcessStringWithSpecialOperation.py
class ProcessStringWithSpecialOperation(object):
    def processStr(self, s):
        """
        :type s: str
        :rtype: str
        """
        result = []

        for ch in s:
            if 'a' <= ch <= 'z':
                result.append(ch)

            elif ch == '*':
                if result:
                    result.pop()

            elif ch == '#':
                result.extend(result)

            elif ch == '%':
                result.reverse()

        return ''.join(result)
sol = ProcessStringWithSpecialOperation()

s = "a#b%*"
print("Input:", s)
print("Output:", sol.processStr(s))